feature/settings/src/main/java/com/google/jetpackcamera/settings/ui/SettingsComponents.kt

```diff
@@ -361,40 +400,28 @@ private fun getStabilizationStringRes(
 @Composable
 fun StabilizationSetting(
     stabilizationUiState: StabilizationUiState,
-    currentTargetFps: Int,
-    currentPreviewStabilization: Stabilization,
-    currentVideoStabilization: Stabilization,
     setVideoStabilization: (Stabilization) -> Unit,
     setPreviewStabilization: (Stabilization) -> Unit,
     modifier: Modifier = Modifier
 ) {
-    // if the preview stabilization was left ON and the target frame rate was set to 15,
-    // this setting needs to be reset to OFF
-    LaunchedEffect(key1 = currentTargetFps, key2 = currentPreviewStabilization) {
-        if (currentTargetFps == FPS_15 &&
-            currentPreviewStabilization == Stabilization.ON
-        ) {
-            setPreviewStabilization(Stabilization.UNDEFINED)
-        }
-    }
     // entire setting disabled when no available fps or target fps = 60
     // stabilization is unsupported >30 fps
     BasicPopupSetting(
         modifier = modifier,
         title = stringResource(R.string.video_stabilization_title),
         leadingIcon = null,
-        enabled = stabilizationUiState.settingEnabledState,
-        description = if (stabilizationUiState.settingEnabledState.isEnabled) {
+        enabled = stabilizationUiState is StabilizationUiState.Enabled,
+        description = if (stabilizationUiState is StabilizationUiState.Enabled) {
```

Comment:
nit: use `when (stabilizationUiState)` here so you get implicit type conversion and don't need to use explicit casting.

Reply:
This applies here and to the rest of this file.